src/Microsoft.AspNet.Mvc.Core/ModelBindingHelper.cs

```diff
@@ -306,16 +307,18 @@ public static class ModelBindingHelper
             };
 
             var modelBindingResult = await modelBinder.BindModelAsync(modelBindingContext);
-            if (modelBindingResult != null)
+            if (modelBindingResult != null && modelBindingResult.IsModelSet)
             {
                 var modelExplorer = new ModelExplorer(metadataProvider, modelMetadata, modelBindingResult.Model);
                 var modelValidationContext = new ModelValidationContext(modelBindingContext, modelExplorer);
-                objectModelValidator.Validate(
-                    modelValidationContext,
-                    new ModelValidationNode(prefix, modelBindingContext.ModelMetadata, modelBindingResult.Model)
-                    {
-                        ValidateAllProperties = true
-                    });
+
+                var validationNode = modelBindingResult.ValidationNode;
+                Debug.Assert(
+                    validationNode != null,
+                    "ValidationNode should never be null in a successful ModelBindingResult.");
```

Comment:
You can pass in a model binder directly to `TryUpdateModel` so it seems a little more likely than usual :laughing:. Normally I'm not paranoid about these kinds of things, but this seems very possible.